### PR TITLE
(PCP-789) Add input_method: powershell for .ps1 files

### PR DESCRIPTION
**Pull request by MikaelSmith**
By default invokes .ps1 files (and any that request input_method: powershell) with a shim that parses JSON into named params. Includes acceptance tests.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
I've run this through testing on Windows 2012r2. I'll try the same on 2008r2 (PowerShell 2) after lunch.

File: acceptance/files/complex-output [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+key: value
+* ArgArrayOfHashes (hashtable[]):
+name: puppet
+version_requirement: >= 3.7.0 < 6.0.0
+name: pe
+version_requirement: >= 5.0.0
+* ArgBool (bool):
+False
+* ArgDateTime (datetime):
+9/8/2017 2:23:59 PM
```
Comment (MikaelSmith) [inline acceptance/files/complex-output]: This might need some munging eventually, on PowerShell 6 on my Mac this was printing `9/8/17 2:23:59 PM`.

**Comment by MikaelSmith** [comment]
PowerShell 2 fails, working on debugging what's wrong in the shim.

**Comment by MikaelSmith** [comment]
Ok, I think this is ready to go. It's passed various acceptance test runs on el-7, Windows 2008r2 and 2012r. I'll kick off an ad-hoc build because I've done some refactoring.

**Comment by MikaelSmith** [comment]
Test run at https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/ad-hoc/job/platform_puppet-agent_init-van-int_suite-daily-ad-hoc/138.

File: exe/PowershellShim.ps1 [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env powershell
+#requires -version 2.0
+
+try {
+  $here = (Split-Path -Parent $MyInvocation.MyCommand.Path)
+  . $here\PowershellShim-Helper.ps1
+
+  $scriptArgsHash = Get-ContentAsJson ($input -join "")
+
+  Write-Debug "`nTask-shim deserialized arguments to Hashtable:`n"
```
Comment (nicklewis) [inline exe/PowershellShim.ps1]: Do we want these `Write-Debug` lines?

Comment (MikaelSmith) [reply exe/PowershellShim.ps1]: They should do nothing normally. They're maybe useful in debugging the shim, but probably not that useful.

**Comment by MikaelSmith** [comment]
Passed acceptance at https://jenkins-master-prod-1.delivery.puppetlabs.net/view/puppet-agent/view/ad-hoc/job/platform_puppet-agent_intn-van-sys_suite-daily-pxp-agent-ad-hoc/27/.

**Comment by nicklewis** [comment]
This looks reasonable to me. I haven't tested it out though.

**Pull request merged**